openedx/core/djangoapps/user_api/management/commands/create_user_gdpr_testing.py

```diff
@@ -0,0 +1,168 @@
+        """
+        Execute the command.
+        """
+
+        username = options['username'] if options['username'] else 'gdpr_test_user_bla'
+        email = options['email'] if options['email'] else 'gdpr_test_user_bla@example.com'
+
+        user, __ = User.objects.get_or_create(
+            username=username,
+            email=email,
+            first_name="GDPR",
+            last_name="Test",
+            is_active=True,
+            password='gdpr test password'
+        )
+
+        # UserProfile
+        profile_image_uploaded_date = datetime(2018, 5, 3, tzinfo=UTC)
+        user_profile, __ = UserProfile.objects.get_or_create(
+            user=user
+        )
+        user_profile_info = {
+            'name': 'gdpr test name',
+            'meta': 'gdpr test meta',
+            'location': 'gdpr test location',
+            'year_of_birth': 1950,
+            'gender': 'gdpr test gender',
+            'mailing_address': 'gdpr test mailing address',
+            'city': 'Boston',
+            'country': 'US',
+            'bio': 'gdpr test bio',
+            'profile_image_uploaded_at': profile_image_uploaded_date
+        }
+        for field, value in user_profile_info.items():
+            setattr(user_profile, field, value)
+        user_profile.save()
+
+        # Profile images
+        with make_image_file() as image_file:
+            create_profile_images(
+                image_file,
+                {10: "ten.jpg"}
+            )
+
+        # DataSharingConsent
+        enterprise_customer, __ = EnterpriseCustomer.objects.get_or_create(
+            name='test gdpr enterprise customer',
+            active=True,
+            branding_configuration=None,
+            catalog=None,
+            enable_audit_enrollment=False,
+            enable_data_sharing_consent=False,
+            enforce_data_sharing_consent='at_enrollment',
+            replace_sensitive_sso_username=True,
+            site_id=1
+        )
+
+        DataSharingConsent.objects.get_or_create(
+            username=username,
+            enterprise_customer_id=enterprise_customer.uuid
+        )
+
+        # Sapsf data transmission
+        enterprise_customer_user, __ = EnterpriseCustomerUser.objects.get_or_create(
+            user_id=user.id,
+            enterprise_customer_id=enterprise_customer.uuid
+        )
+        audit, __ = EnterpriseCourseEnrollment.objects.get_or_create(
+            enterprise_customer_user=enterprise_customer_user
+        )
+        SapSuccessFactorsLearnerDataTransmissionAudit.objects.get_or_create(
+            enterprise_course_enrollment_id=audit.id,
+            completed_timestamp=10
+        )
+
+        # PendingEnterpriseCustomerUser
+        PendingEnterpriseCustomerUser.objects.get_or_create(
+            user_email=user.email,
+            enterprise_customer_id=enterprise_customer.uuid
+        )
+
+        # EntitlementSupportDetail
+        fake_course_uuid = uuid4().hex
```

Comment:
You might want to have an optional course_id argument, so someone could manage testing against a course they have set up locally.